### PR TITLE
Chiggs upstream patches

### PR DESCRIPTION
**Pull request by imphil**
Rebased version of #708

### CODE REVIEW

File: cocotb/drivers/amba.py [hunk 1]
```diff
@@ -184,6 +184,8 @@ def read(self, address, sync=True):
 
         raise ReturnValue(data)
 
+    def __len__(self):
+        return 2**len(self.bus.ARADDR)
```
Comment (eric-wieser) [inline cocotb/drivers/amba.py]: As mentioned in the original CR - this doesn't make a whole lot of sense to me, and seems like an abuse of `len`

Comment (imphil) [reply cocotb/drivers/amba.py]: Would you mind filing a new issue to find a better solution for that? The PR and the associated discussion have gotten way too long with many unrelated changed in flight.

File: makefiles/simulators/Makefile.icarus [hunk 2]
```diff
@@ -54,6 +54,8 @@ endif
 
 BUILD_VPI=1
 
+COMPILE_ARGS += -g2012 # Default to latest SystemVerilog standard
```
Comment (eric-wieser) [inline makefiles/simulators/Makefile.icarus]: I think this should be conditional on having a new enough version of `iverilog` - this breaks ubuntu 16, which comes with 9.7

**Comment by imphil** [comment]
This PR is already closed and merged, let's please have follow-up discussions in new issues. I've filed https://github.com/potentialventures/cocotb/issues/781 for that.

**Pull request merged**